### PR TITLE
Reserved ip

### PR DESCRIPTION
**Pull request by RealHarshThakur**
* PR provides a better structure for subcommand
* Adds IP subcommand
* Changes instance public-ip command functionality. Users can now enable/disable public IP on an instance

### CODE REVIEW

File: cmd/ip/ip_assign.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+				os.Exit(1)
+			} else if errors.Is(err, civogo.MultipleMatchesError) {
+				utility.Error("sorry we found more than one IP with that value in your account")
+				os.Exit(1)
+			} else {
+				utility.Error("%s", err)
+			}
+		}
+
+		instance, err := client.FindInstance(args[1])
```
Comment (haardikdharma10) [inline cmd/ip/ip_assign.go]: Wondering how is this working? We always pass instance name/ID with the `instance` flag and not as an argument right? 🤔

File: cmd/ip/ip.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/spf13/cobra"
+)
+
+var name, instance string
+
+// IPCmd manages reserved IPs
+var IPCmd = &cobra.Command{
+	Use:     "ip",
+	Aliases: []string{"ips"},
```
Comment (haardikdharma10) [inline cmd/ip/ip.go:14]: We can have more aliases - {ip-address, ip-addresses, reserved-ip, reserved-ips}

File: cmd/ip/ip_unassign.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/civo/civogo"
+	"github.com/civo/cli/common"
+	"github.com/civo/cli/config"
+	"github.com/civo/cli/utility"
+	"github.com/spf13/cobra"
+)
+
+var ipUnassignCmd = &cobra.Command{
+	Use:     "Unassign",
+	Aliases: []string{"attach"},
```
Comment (haardikdharma10) [inline cmd/ip/ip_unassign.go]: unattach

File: cmd/ip/ip_assign.go (hunk 1 above)
Comment (RealHarshThakur) [reply cmd/ip/ip_assign.go]: Fixed

File: cmd/ip/ip.go (hunk 2 above)
Comment (RealHarshThakur) [reply cmd/ip/ip.go:14]: Feels a little counter-intuitive to me that someone would type in the longer name

**Comment by vishalanarase** [comment]
LGTM

**Review by vishalanarase — APPROVED** [review]

**Pull request merged**